### ISSUE TITLE
Castro reactions fail with ROCm > 5.3.0

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#2569]
Using this issue to track problems with more recent versions of ROCm

**Comment by zingale** [comment]
To build with ROCm 5.7.0 on Frontier:

```
module load cpe/23.09
module load rocm/5.7.0
module load PrgEnv-gnu craype-accel-amd-gfx90a cray-mpich
```

**Comment by zingale** [comment]
Running the 2D subchandra problem on 4 nodes / 32 GPUs:

* We run fine if compiled with `DEBUG=TRUE` (although the compilng / linking takes a Hubble time)
* We crash with problems in the reaction integration right away (step 1) if compiled normally.

**Comment by zingale** [comment]
More precisely, the burning fails and the step is thrown out and we need to subcycle.

**Comment by zingale** [comment]
The ``reacting_convergence`` problem works fine, it seems.  This could be because of Strang vs. SDC, or the type of network.

**Comment by zingale** [comment]
okay, with subchandra and Strang, and ROCm 5.7.0 I now get:

```
... Entering burner on level 0 and doing half-timestep of burning.

:0:rocdevice.cpp            :2692: 142346779052 us: [pid:90391 tid:0x7fffd6530700] Callback: Queue 0x7ffe91600000 aborting with 
error : HSA_STATUS_ERROR_MEMORY_APERTURE_VIOLATION: The agent attempted to access memory beyond the largest legal address. code:
 0x29
SIGABRT
Memory access fault by GPU node-8 (Agent handle: 0x2975b60) on address 0x800033773000. Reason: Unknown.
SIGABRT
See Backtrace.24 file for details
See Backtrace.7 file for details
```

**Comment by BenWibking** [comment]
Is this issue we're seeing when running problems with Microphysics in Quokka possibly related: https://github.com/quokka-astro/quokka/issues/394?

It only happens on AMDGPU, and only when we use reaction networks.

**Comment by zingale** [comment]
We run without problem with ROCm 5.3.0 and reactions.  So I am not sure if it is the same.

**Comment by BenWibking** [comment]
> We run without problem with ROCm 5.3.0 and reactions. So I am not sure if it is the same.

Hmm, that makes it probably unrelated. Have you happened to run with ROCm 5.2.3? We only have access to this version on our machine.

**Comment by zingale** [comment]
possibly related:
https://github.com/AMReX-Codes/amrex/issues/3623

**Comment by zingale** [comment]
this is fixed in #2749, and is most likely a ROCm compiler bug

**Comment by yut23** [comment]
I just tested the failing subchandra setup with the hipcc force-inlining turned off and without #2749, and that also resolves the memory error.
